flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -484,27 +484,15 @@ public final void invoke() throws Exception {
 
 	private void runMailboxLoop() throws Exception {
 		try {
-			try {
-				mailboxProcessor.runMailboxLoop();
-			}
-			catch (WrappingRuntimeException wrappingException) {
-				Throwable unwrapped = wrappingException.unwrap();
-				if (unwrapped instanceof Exception) {
-					throw (Exception) unwrapped;
-				}
-				else {
-					throw wrappingException;
-				}
-			}
-		}
-		catch (InterruptedException e) {
-			if (!canceled) {
-				Thread.currentThread().interrupt();
-				throw e;
-			}
+			mailboxProcessor.runMailboxLoop();
 		}
 		catch (Exception e) {
-			if (canceled) {
+			if (ExceptionUtils.findThrowable(e, InterruptedException.class).isPresent()) {
```

Comment:
We could add a simple test coverage for this, for example by extending `SourceStreamTaskTest#testInterruptedNotSwallowed` with a boolean parameter, controlling whether the exception is thrown wrapped or not (similar pattern as  `SourceStreamTaskTest#testCancellationWithSourceBlockedOnLock(boolean, boolean)`.